### PR TITLE
Added a few changes to "soften" the read function from the Cursor:

### PR DESCRIPTION
**Pull request by kodeschreiber**
- Added a "contains" function to allow a soft-check for existing indexes
- Implemented the contains function for the RowIndex implementations
- Removed the debug_asset statements that were originally causing the panics

### CODE REVIEW

File: src/cursor.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ pub trait RowIndex: std::fmt::Debug {
     ///
     /// The first column has index 0.
     fn index(self, row: &Row) -> usize;
+    fn contains(&self, row: &Row) -> bool;
```
Comment (IvanUkhov) [inline src/cursor.rs:37]: Please move it up and add a docstring.

File: src/cursor.rs [hunk 2]
```diff
@@ -154,6 +155,10 @@ impl Row {
     where
         U: RowIndex,
     {
+        if !column.contains(self) {
```
Comment (IvanUkhov) [inline src/cursor.rs:158]: Please remove this one.

File: src/cursor.rs [hunk 3]
```diff
@@ -185,27 +194,34 @@ where
...
+        } else {
+            panic!("Index does not exist in given Row");
+        }
     }
 }
 
 impl RowIndex for &str {
     #[inline]
     fn index(self, row: &Row) -> usize {
-        debug_assert!(
```
Comment (IvanUkhov) [inline src/cursor.rs:195]: Please recover.

File: src/cursor.rs [hunk 4]
```diff
@@ -185,27 +194,34 @@ where
...
+
+    fn contains(&self, row: &Row) -> bool {
+      row.column_mapping.contains_key(*self)
+    }
 }
 
 impl RowIndex for usize {
     #[inline]
     fn index(self, row: &Row) -> usize {
-        debug_assert!(self < row.values.len(), "the index is out of range");
```
Comment (IvanUkhov) [inline src/cursor.rs:206]: Please recover.

File: src/cursor.rs [hunk 5]
```diff
@@ -185,27 +194,34 @@ where
...
         self
     }
+
+    fn contains(&self, row: &Row) -> bool {
```
Comment (IvanUkhov) [inline src/cursor.rs:222]: Please put it before `index`.

File: src/cursor.rs [hunk 6]
```diff
@@ -185,27 +194,34 @@ where
...
     #[inline]
     fn index(self, row: &Row) -> usize {
-        debug_assert!(
-            row.column_mapping.contains_key(self),
-            "the index is out of range",
-        );
         row.column_mapping[self]
     }
+
+    fn contains(&self, row: &Row) -> bool {
```
Comment (IvanUkhov) [inline src/cursor.rs:211]: Please put it before index.

File: src/cursor.rs [hunk 7]
```diff
@@ -185,27 +194,34 @@ where
     type Output = Value;
 
     fn index(&self, index: T) -> &Value {
-        &self.values[index.index(self)]
+        if index.contains(self) {
```
Comment (IvanUkhov) [inline src/cursor.rs:197]: Please recover how it used to be. It is the same effect.

**Review by IvanUkhov — COMMENTED** [review]
I am all right with adding `contains`. But I would not want to have the other changes for now.

**Comment by kodeschreiber** [comment]
Those changes are fine with me. I'm going to open a new PR with the suggestions applied to your latest commits on main.

**Pull request closed without merging**